### PR TITLE
add lxd-profile charm

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Adding an lxd-profile charm to facilitate and test the lxd profile updates by charms.  Blacklist items are commented out for future use.

## QA steps

deploy the charm locally.

### CODE REVIEW

**Comment by hmlanigan** [comment]
[create sample charm with a lxd profile tweak in the charm](https://trello.com/c/2pl44PMg/489-create-sample-charm-with-a-lxd-profile-tweak-in-the-charm)

File: testcharms/charm-repo/quantal/lxd-profile/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: lxd-profile
+summary: start a juju machine with an lxd profile
```
Comment (SimonRichardson) [inline testcharms/charm-repo/quantal/lxd-profile/metadata.yaml]: nit "a"

File: testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#name: juju-default
+description: lxd profile for testing, black list items grouped commented out
+config:
+#
+# allowed config
+#
+  security.nesting: "true"
+  security.privileged: "true"
```
Comment (SimonRichardson) [inline testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml]: This one would require a restart, so we should consider how we highlight this :thinking:

Comment (hmlanigan) [reply testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml]: agreed, a question/comment in the discourse spec?

**Comment by hmlanigan** [comment]
I will put up a pr for the charm which will fail, for use with --force.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**